esp_flasher/__main__.py

```
@@ -241,11 +241,74 @@
+        return os.path.join(venv, "bin", "python3")
+    return sys.executable
+
+def is_admin():
+    """
+    Check for admin rights (nessessary on linux)
+    """
+    if sys.platform == 'win32':
+        try:
+            import ctypes
+            return ctypes.windll.shell32.IsUserAdmin() #true when UAC_Admin
+        except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)